### PR TITLE
choose-rust 1.2.0 (new formula)

### PR DESCRIPTION
**Pull request by crclark96**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Adding a formula for the `choose` utility from [theryangeary/choose](https://github.com/theryangeary/choose)
(the formula is named choose-rust since another, seemingly unmaintained, project has the same name)

I added conflict statements to this formula but not the others - let me know if you'd like me to add them

### CODE REVIEW

**Comment by vladimyr** [comment]
> I added conflict statements to this formula but not the others - let me know if you'd like me to add them

You should add conflict markers on both sides inside same PR. So feel free to update other formulas too.

File: Formula/choose-rust.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+  conflicts_with "choose", :because => "both install a `choose` binary"
+  conflicts_with "choose-gui", :because => "both install a `choose` binary"
+
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    system "choose", "--version"
```
[the remark this answers is not in the document]
Comment (SMillerDev) [reply Formula/choose-rust.rb:17]: [`pipe_output`](https://rubydoc.brew.sh/Homebrew/Assertions.html#pipe_output-instance_method) might be a lot easier there

[the remark this answers is not in the document]
Comment (vladimyr) [reply Formula/choose-rust.rb:17]: Indeed 👍 
```suggestion
    input = "foo,  foobar,bar, baz"
    assert_equal "foobar bar", pipe_output("#{bin}/choose -f ',\\s*' 1..=2", input).strip
```

[the remark this answers is not in the document]
Comment (crclark96) [reply Formula/choose-rust.rb]: Thanks for the great suggestions - integrated

**Comment by crclark96** [comment]
Tests improved and conflict statements added to the other formulae

Let me know if there's anything else

**Review by chenrui333 — DISMISSED** [review]
Nice work!

**Review by SMillerDev — APPROVED** [review]

**Comment by SMillerDev** [comment]
Thanks @crclark96 ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Pull request closed without merging**